front/src/common/Collapsable.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useCallback, useEffect, useState, type FC, type HTMLAttributes } from 'react';
+
+import { ChevronDown, ChevronUp } from '@osrd-project/ui-icons';
+import cx from 'classnames';
+import { omit } from 'lodash';
+import { useTranslation } from 'react-i18next';
+
+interface CollapsableProps extends Omit<HTMLAttributes<unknown>, 'children' | 'collapsed'> {
+  children: [JSX.Element, JSX.Element];
+  collapsed: boolean;
+}
+const Collapsable: FC<CollapsableProps> = (props) => {
+  const { children, collapsed, className } = props;
+  const { t } = useTranslation('common/common');
+  const [isCollapsed, setIsCollapsed] = useState(collapsed);
+
+  const toggle = useCallback(() => {
+    setIsCollapsed((prev) => !prev);
+  }, []);
```

Comment:
What is the difference between doing this or do this in the button's `onClick` ?

```suggestion
  onClick={() => {
    setIsCollapsed((prev) => !prev);
  }}
```

Reply:
If you do that, at every render you redefine the function, so you change its ref, so you retrigger the render of the children component.

I'm ok that here it's not really needed, the function is really tiny, and it's a `button` component.
But it cost nothing to do it like that.